### DISCUSSION TITLE
data

### DISCUSSION THREAD

**Discussion by LELEX76-arch** [micropython/micropython#11769] [ESP32]
I want to programme a ESP 32 for to make a data as sound , for example a external data from .csv  of weather

**Comment by sosi-deadeye** [comment]
To output sound, you could use I2S.
Example device which works: https://www.amazon.de/gp/product/B09NXKPZ8N
![61X-l4cW8LL _SL1200_](https://github.com/micropython/micropython/assets/3900541/66d2a7d7-3947-414d-adbb-0f87a6b82b0d)


Docs: https://docs.micropython.org/en/latest/library/machine.I2S.html

Then you need to save the data on the internal flash.
I use a SDCard to store plain WAVE-Files (PCM - Pulse Coded Modulation) on it.
